tokio-net/src/addr.rs

```diff
@@ -31,6 +31,18 @@ impl sealed::ToSocketAddrsPriv for SocketAddr {
     }
 }
 
+/// `resolve_host` takes `ToSocketAddrs` and returns the first entry it resolves.
+pub async fn resolve_host<T: ToSocketAddrs>(source: T) -> Result<SocketAddr, io::Error> {
```

Comment:
you do not allow to retrieve all possible addresses, is that intentional?
Maybe return `T::Iter` instead?